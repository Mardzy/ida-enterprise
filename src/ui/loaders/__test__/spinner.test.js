import SpinnerWithOverlay from "../spinner";

describe("The spinner with overlay", () => {

  it("renders a spinner", () => {
    const wrapper = mount(
      <SpinnerWithOverlay/>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders a spinner with text", () => {
    const wrapper = mount(
      <SpinnerWithOverlay>Loading content</SpinnerWithOverlay>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

});