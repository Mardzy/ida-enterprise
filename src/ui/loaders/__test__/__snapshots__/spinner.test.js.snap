// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The spinner with overlay renders a spinner 1`] = `
<SpinnerWithOverlay>
  <Dimmer
    active={true}
    inverted={true}
  >
    <DimmerInner
      active={true}
      inverted={true}
    >
      <Ref
        innerRef={
          Object {
            "current": <div
              class="ui active transition visible inverted dimmer"
              style="display: flex;"
            >
              <div
                class="content"
              >
                <div
                  class="ui inverted loader"
                />
              </div>
            </div>,
          }
        }
      >
        <RefFindNode
          innerRef={
            Object {
              "current": <div
                class="ui active transition visible inverted dimmer"
                style="display: flex;"
              >
                <div
                  class="content"
                >
                  <div
                    class="ui inverted loader"
                  />
                </div>
              </div>,
            }
          }
        >
          <div
            className="ui active transition visible inverted dimmer"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <Loader
                inverted={true}
              >
                <div
                  className="ui inverted loader"
                />
              </Loader>
            </div>
          </div>
        </RefFindNode>
      </Ref>
    </DimmerInner>
  </Dimmer>
</SpinnerWithOverlay>
`;

exports[`The spinner with overlay renders a spinner with text 1`] = `
<SpinnerWithOverlay>
  <Dimmer
    active={true}
    inverted={true}
  >
    <DimmerInner
      active={true}
      inverted={true}
    >
      <Ref
        innerRef={
          Object {
            "current": <div
              class="ui active transition visible inverted dimmer"
              style="display: flex;"
            >
              <div
                class="content"
              >
                <div
                  class="ui inverted text loader"
                >
                  Loading content
                </div>
              </div>
            </div>,
          }
        }
      >
        <RefFindNode
          innerRef={
            Object {
              "current": <div
                class="ui active transition visible inverted dimmer"
                style="display: flex;"
              >
                <div
                  class="content"
                >
                  <div
                    class="ui inverted text loader"
                  >
                    Loading content
                  </div>
                </div>
              </div>,
            }
          }
        >
          <div
            className="ui active transition visible inverted dimmer"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <Loader
                inverted={true}
              >
                <div
                  className="ui inverted text loader"
                >
                  Loading content
                </div>
              </Loader>
            </div>
          </div>
        </RefFindNode>
      </Ref>
    </DimmerInner>
  </Dimmer>
</SpinnerWithOverlay>
`;
